src/gleam/iterator.gleam

```diff
@@ -1574,6 +1574,33 @@ pub fn length(over iterator: Iterator(e)) -> Int {
   |> do_length(0)
 }
 
+/// Counts the number of elements in the given iterator satisfying the given
+/// predicate.
+///
+/// This function has to traverse the entire iterator to count its elements,
+/// so it runs in linear time.
+///
+/// ## Examples
+///
+/// ```gleam
+/// empty() |> count(int.is_odd)
+/// // -> 0
+/// ```
+///
+/// ```gleam
+/// from_list([1, 2, 3, 4]) |> count(int.is_odd)
+/// // -> 2
+/// ```
+///
+pub fn count(over iterator: Iterator(e), where predicate: fn(e) -> Bool) -> Int {
```

Comment:
I'm not sure what the use case is for this function. It's the same as `iter |> iterator.filter(f) |> iterator.length`, as opposed to with lists where it has different performance characteristics.

Reply:
That is true, there is no performance benefit to `iter |> count(f)` over `iter |> filter(f) |> length`. It does however provide a simplified API (like many other functions are just simplifications on `fold`).

Having `list.count` available but not `iterator.count` would be a (minor) confusing discrepancy. If it exists for `list` I would assume it also exists for `iterator`. That's of course just my opinion.

But it's up to you, I can remove `iterator.count` again if you don't like its addition 😄

Reply:
@lpil I moved this discussion to the issue (#592).